### DISCUSSION TITLE
How can I add camouflage pattern to the car body?

### DISCUSSION THREAD

**Discussion by keode** [carla-simulator/carla#5695] [Q&A]
I would like to add a mosaic-like camouflage pattern to the car body, for the purpose of evaluating adversarial attack on an object detector (YOLO for example). A camouflaged car would look like [this](https://imgur.com/a/Wuoqgww).

My gut-feeling tells me that I have to compile the Unreal Engine that Carla specially makes, open the car asset in Unreal editor and customize the exterior material of the car (following the tutorial [here](https://carla.readthedocs.io/en/0.9.13/tuto_A_material_customization/#exterior-properties)). Is that correct?

As I need to evaluate many camouflage patterns, I would also like to know if there is a Python API to change the cover image of the car body on the fly, instead of manually editing the car asset in Unreal editor?

**Comment by guotuLuo** [comment]
hey,bro,have you solved the problem, i meet the similiar problem with you
